psi-java/src/main/kotlin/io/spine/tools/psi/java/TrimmingPsiFactory.kt

```diff
@@ -0,0 +1,164 @@
+ *
+ * [PsiElementFactory.createClassFromText] also left intact because the created class is nested
+ * into outer `Dummy` class. So, the passed whitespaces go to definition of `Dummy`.
+ */
+@Suppress("TooManyFunctions")
+internal class TrimmingPsiFactory(
+    private val delegate: PsiElementFactory
+) : PsiElementFactory by delegate {
+
+    override fun createFieldFromText(p0: String, p1: PsiElement?): PsiField =
+        delegate.createFieldFromText(p0.trimStart(), p1)
+
+    override fun createMethodFromText(p0: String, p1: PsiElement?, p2: LanguageLevel?): PsiMethod =
+        delegate.createMethodFromText(p0.trimStart(), p1, p2)
+
+    override fun createMethodFromText(p0: String, p1: PsiElement?): PsiMethod =
+        delegate.createMethodFromText(p0.trimStart(), p1)
+
+    override fun createParameterFromText(p0: String, p1: PsiElement?): PsiParameter =
+        delegate.createParameterFromText(p0.trimStart(), p1)
+
+    override fun createRecordHeaderFromText(p0: String, p1: PsiElement?): PsiRecordHeader =
+        delegate.createRecordHeaderFromText(p0.trimStart(), p1)
+
+    override fun createResourceFromText(p0: String, p1: PsiElement?): PsiResourceVariable =
+        delegate.createResourceFromText(p0.trimStart(), p1)
+
+    override fun createTypeFromText(p0: String, p1: PsiElement?): PsiType =
+        delegate.createTypeFromText(p0.trimStart(), p1)
+
+    override fun createTypeElementFromText(p0: String, p1: PsiElement?): PsiTypeElement =
+        delegate.createTypeElementFromText(p0.trimStart(), p1)
+
+    override fun createReferenceFromText(p0: String, p1: PsiElement?): PsiJavaCodeReferenceElement =
+        delegate.createReferenceFromText(p0.trimStart(), p1)
+
+    override fun createCodeBlockFromText(p0: String, p1: PsiElement?): PsiCodeBlock =
+        delegate.createCodeBlockFromText(p0.trimStart(), p1)
+
+    override fun createStatementFromText(p0: String, p1: PsiElement?): PsiStatement =
+        delegate.createStatementFromText(p0.trimStart(), p1)
+
+    override fun createExpressionFromText(p0: String, p1: PsiElement?): PsiExpression =
+        delegate.createExpressionFromText(p0.trimStart(), p1)
+
+    // Use `trim()` because this method also prohibits trailing whitespaces.
+    override fun createCommentFromText(p0: String, p1: PsiElement?): PsiComment =
+        delegate.createCommentFromText(p0.trim(), p1)
+
+    override fun createTypeParameterFromText(p0: String, p1: PsiElement?): PsiTypeParameter =
+        delegate.createTypeParameterFromText(p0.trimStart(), p1)
+
+    override fun createAnnotationFromText(p0: String, p1: PsiElement?): PsiAnnotation =
+        delegate.createAnnotationFromText(p0.trimStart(), p1)
+
+    override fun createEnumConstantFromText(p0: String, p1: PsiElement?): PsiEnumConstant =
+        delegate.createEnumConstantFromText(p0.trimStart(), p1)
+
+    // Use `trim()` because this method also prohibits trailing whitespaces.
```

Comment:
Please have a proper function-level comment instead of this one: it should sound answering to the question "What it is" or "What it does", and be formatted properly.

It's here, and also, it's a few lines above.